### ISSUE TITLE
Simplify save_graph_xml output

### ISSUE THREAD

**Issue by davidmurray** [gboeing/osmnx#915]
**Do not delete the following template. Fill it out completely.**

  - [x] I have read the OSMnx [contributing guidelines](../blob/main/CONTRIBUTING.md)
  - [x] I have read the OSMnx [documentation](https://osmnx.readthedocs.io/) and [usage examples](https://github.com/gboeing/osmnx-examples)
  - [x] I have searched the open/closed issues and pull requests for others similar to mine

**Is your feature proposal related to a problem?**

The `save_graph_xml` function adds a lot of (unnecessary?) data to the resulting XML file:
https://github.com/gboeing/osmnx/blob/586cf2651af80c2cdb1c6e30b94d8176beea7c4a/osmnx/osm_xml.py#L207-L212

The resulting XML file has a lot of extra fields compared to regular OSM XML files :
```xml
<?xml version="1.0"?>
<osm version="0.6" generator="OSMnx">
  <node id="0" timestamp="2017-01-01T00:00:00Z" uid="1" user="osmnx" version="1" changeset="1" lat="45.5571316" lon="-73.5963673">
    <tag k="highway" v="nan"/>
  </node>
  <node id="1" timestamp="2017-01-01T00:00:00Z" uid="1" user="osmnx" version="1" changeset="1" lat="nan" lon="nan">
    <tag k="highway" v="nan"/>
  </node>
  <node id="2" timestamp="2017-01-01T00:00:00Z" uid="1" user="osmnx" version="1" changeset="1" lat="nan" lon="nan">
    <tag k="highway" v="nan"/>
  </node>
  <node id="3" timestamp="2017-01-01T00:00:00Z" uid="1" user="osmnx" version="1" changeset="1" lat="45.5575365" lon="-73.5889391">
    <tag k="highway" v="nan"/>
  </node>
  ...
</osm>
```

In addition, what's the point of the timestamp if it's always the same date in 2017? 

**Describe the solution you'd like to propose**

Either:
1. Adding a parameter to `save_graph_xml` to decide whether or not these extra fields should be added to the xml file, or:
2. Removing this code altogether.

Thanks!

**Comment by davidmurray** [comment]
In addition, it seems that the function adds tags even when the values do not exist:
```xml
<node id="2" timestamp="2017-01-01T00:00:00Z" uid="1" user="osmnx" version="1" changeset="1" lat="nan" lon="nan">
    <tag k="highway" v="nan"/>
  </node>
```

![image](https://user-images.githubusercontent.com/545502/210832750-851e67a0-730e-4089-95c2-db53f13f688a.png)

When the values are null/NaN, they shouldn't be written to the xml file in my opinion. Might be worth fixing this as well. :)

**Comment by gboeing** [comment]
Do the `osm_xml_*` values in the `settings` module not allow you to do this as you wish?

**Comment by davidmurray** [comment]
@gboeing Hi,

Yeah, it seems like it can. `ox.settings.osm_xml_node_attrs` allows be to specify which keys I want. However, it doesn't let me specify the values. I.e. it's always these:
```
table["uid"] = "1"
table["user"] = "osmnx"
table["version"] = "1"
table["changeset"] = "1"
table["timestamp"] = "2017-01-01T00:00:00Z"
```

**Comment by mxndrwgrdnr** [comment]
I don't recall exactly, but I recall that at the time I wrote this code I was using a command-line tool (Osmium?) for test that my outputs were valid .osm files. I'm pretty sure that without these fields present I couldn't get my files to validate, or at least the validator would complain.

**Comment by gboeing** [comment]
See #917

**Comment by gboeing** [comment]
Closed by #917

**Comment by gboeing** [comment]
See enhancements in #1135

User can now specify custom values for each attribute by simply setting those attribute values on the graph nodes/edges before saving as XML.
